pkg/cloud/azure/azure.go

```diff
@@ -23,6 +30,24 @@ var (
 	}
 )
 
+var (
+	validAzureCloudNames = map[configv1.AzureCloudEnvironment]bool{
+		configv1.AzurePublicCloud:       true,
+		configv1.AzureUSGovernmentCloud: true,
+		configv1.AzureChinaCloud:        true,
+		configv1.AzureGermanCloud:       true,
+		configv1.AzureStackCloud:        true,
+	}
```

Comment:
I'm guessing this was copy/pasted? Nit, `map[T]struct{}` is more efficient than `map[T]bool`

Reply:
Yep, it is copy/pasted from the existing functionality in the CCO. 

I didn't know that! Out of interest, what makes `map[T]struct{}` faster? Is it significantly faster for small sets, or only large ones?

Reply:
Memory allocation, `struct{}` allocates less memory than `bool`, so the idiom in Go is to use `struct{}` when you need a set semantic